frappe/core/doctype/user/user.json

```diff
@@ -214,9 +214,9 @@
   {
    "allow_in_quick_entry": 1,
    "fieldname": "role_profile_name",
-   "fieldtype": "Link",
+   "fieldtype": "Table MultiSelect",
```

Comment:
We shouldn't reuse same field. It also breaks semantics. It should be called `role_profiles`. 

If we add a new field then can safely make it backward compatible too.